src/named_tuple.cr

```diff
@@ -38,13 +38,16 @@ struct NamedTuple
   end
 
   # Creates a named tuple from the given hash, with elements casted to the given types.
+  # Here the Int32 | String union is cast to Int32.
   #
   # ```
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2})       # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({"foo" => "world", "bar" => 2})     # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2}).class # => {foo: String, bar: Int64}
-  # ```
+  # num_or_str : Int32 | String
+  # num_or_str = 42
```

Comment:
Ah thanks - thats interesting since:
```crystal
str_or_num : String | Int32
str_or_num = 1_i64 => type must be (Int32 | String), not Int64
```
I will use ``num_or_str = 42.as(Int32 | String)`` thanks